### DISCUSSION TITLE
How can I access the model loaded in the scene?

### DISCUSSION THREAD

**Discussion by yasarkurt** [google/model-viewer#2512] [Q&A]
I need to be able to access the modle to disassemble the model loaded into the scene. Even if I can't access the model, is it possible for me to access its subparts?

**Comment by elalish** [comment]
We're working on expanding our scene-graph API to enable this. Can you give us some details of your use case to consider as we are designing? In the meanwhile, you can hack your way down to the Three.js layer and do anything you want (you'll want to get a bit familiar with our code base), but keep in mind that you'll likely have to deal with breaking changes as we update.

**Comment by yasarkurt** [comment]
Hello Emmett Lalish,

The model I want to make is to give an explosion effect. I want to take the meshes of the added model into an array and update their positions.
As an example,
[cid:ff52f5dc-e597-4ce8-95b5-9c8e2a77d2ea]

Good work
________________________________
Gönderen: Emmett Lalish ***@***.***>
Gönderildi: 25 Haziran 2021 Cuma 18:47
Kime: google/model-viewer ***@***.***>
Bilgi: yasarkurt ***@***.***>; Author ***@***.***>
Konu: Re: [google/model-viewer] How can I access the model loaded in the scene? (#2512)


We're working on expanding our scene-graph API to enable this. Can you give us some details of your use case to consider as we are designing? In the meanwhile, you can hack your way down to the Three.js layer and do anything you want (you'll want to get a bit familiar with our code base), but keep in mind that you'll likely have to deal with breaking changes as we update.

**Reply by elalish** [reply]
For exploded views, I would highly recommend authoring an animation (or animations) in the glTF file, and then selecting and playing one with our existing API. This will make for dramatically less code on your side.

**Comment by yasarkurt** [comment]
You're saying that animating for the explosion will give higher performance?

So if I want to list meshes, is there no model-viewer support for it?
________________________________
Gönderen: Emmett Lalish ***@***.***>
Gönderildi: 25 Haziran 2021 Cuma 20:43
Kime: google/model-viewer ***@***.***>
Bilgi: yasarkurt ***@***.***>; Author ***@***.***>
Konu: Re: [google/model-viewer] How can I access the model loaded in the scene? (#2512)


For exploded views, I would highly recommend authoring an animation (or animations) in the glTF file, and then selecting and playing one with our existing API. This will make for dramatically less code on your side.

**Reply by elalish** [reply]
That's correct, not yet. The purpose of having animations as part of the file is that you have much better tools for authoring them, plus they are then portable to other uses beyond model-viewer as well.

**Comment by yasarkurt** [comment]
Thank you for your help.
Have a nice day.
________________________________
Gönderen: Emmett Lalish ***@***.***>
Gönderildi: 25 Haziran 2021 Cuma 20:51
Kime: google/model-viewer ***@***.***>
Bilgi: yasarkurt ***@***.***>; Author ***@***.***>
Konu: Re: [google/model-viewer] How can I access the model loaded in the scene? (#2512)


That's correct, not yet. The purpose of having animations as part of the file is that you have much better tools for authoring them, plus they are then portable to other uses beyond model-viewer as well.
